### PR TITLE
Rdm 2676

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
RDM-2676: Validate CCD reference number when entered as a CaseLink
    
                [RDM-2676] (https://tools.hmcts.net/jira/browse/RDM-2676).

### CODE REVIEW

File: charts/ccd-data-store-api/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ description: Helm chart for the HMCTS CCD Data Store
 name: ccd-data-store-api
 apiVersion: v1
 home: https://github.com/hmcts/ccd-data-store-api
-version: 1.6.2
+version: 1.6.3
 maintainers:
```
Comment (Thor-tech-of-metal) [inline charts/ccd-data-store-api/Chart.yaml:6]: I need to pull latest dev .

File: charts/ccd-data-store-api/values.yaml [hunk 2]
```diff
@@ -25,7 +25,7 @@ java:
 
     CCD_DRAFT_TTL_DAYS: 180
 
-    DATA_STORE_S2S_AUTHORISED_SERVICES: ccd_data,ccd_gw,ccd_ps,probate_backend,divorce_ccd_submission,sscs,sscs_bulkscan,cmc,cmc_claim_store,cmc_claim_external_api,jui_webapp,pui_webapp,bulk_scan_orchestrator,fpl_case_service,iac,finrem_ccd_data_migrator,finrem_case_orchestration,employment_tribunals,ethos_repl_service,ccpay_bubble,ctsc_work_allocation,em_ccd_orchestrator,xui_webapp,bulk_scan_payment_processor,pcq_consolidation_service,em_npa_app,ecm_consumer,aac_manage_case_assignment
+    DATA_STORE_S2S_AUTHORISED_SERVICES: ccd_data,ccd_gw,ccd_ps,probate_backend,divorce_ccd_submission,sscs,sscs_bulkscan,cmc,cmc_claim_store,cmc_claim_external_api,jui_webapp,pui_webapp,bulk_scan_orchestrator,fpl_case_service,iac,finrem_ccd_data_migrator,finrem_case_orchestration,employment_tribunals,ethos_repl_service,ccpay_bubble,ctsc_work_allocation,em_ccd_orchestrator,xui_webapp,bulk_scan_payment_processor,pcq_consolidation_service,em_npa_app,ecm_consumer,aac_manage_case_assignment,unspec_service
```
Comment (Thor-tech-of-metal) [inline charts/ccd-data-store-api/values.yaml:29]: I need to pull latest dev .

File: src/main/java/uk/gov/hmcts/ccd/domain/types/TextCaseReferenceCaseLinkValidator.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import uk.gov.hmcts.ccd.endpoint.exceptions.ResourceNotFoundException;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.Collections;
+import java.util.List;
+import static uk.gov.hmcts.ccd.domain.types.TextValidator.checkRegex;
+
+@Named("CaseLinkValidator")
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/types/TextCaseReferenceCaseLinkValidator.java:15]:
```suggestion
@Named("TextCaseReferenceCaseLinkValidator")
```

File: src/main/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidator.java [hunk 4]
```diff
@@ -98,14 +101,21 @@ public CaseDataValidator(final List<BaseTypeValidator> validators) {
...
-                .collect(Collectors.toList()))
-            .orElseThrow(() -> new RuntimeException("System error: No validator found for " + fieldType.getType()));
+            .filter(validator ->
+                validator.getType() == fieldType || isPredefinedTypeBaseValidator(validator, caseFieldDefinition.getFieldTypeDefinition().getId())
+            )
+            .map(baseTypeValidator -> baseTypeValidator.validate(dataFieldId, dataValue, caseFieldDefinition))
+            .flatMap(validationResults -> validationResults.stream()
+                .map(validationResult -> new ValidationResult(validationResult.getErrorMessage(), fieldIdPrefix + validationResult.getFieldId()))
+            ).collect(Collectors.toList());
+    }
```
Comment (Thor-tech-of-metal) [inline src/main/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidator.java:111]: We can execute  all sorted validators and always take the first elements from the result, that can lead in more testing but it is generic enough

File: src/main/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidator.java [hunk 5]
```diff
@@ -46,7 +46,11 @@ public CaseDataValidator(final List<BaseTypeValidator> validators) {
                 .map(caseDataPair -> caseFieldDefinitions.stream()
                     .filter(caseField -> caseField.getId().equalsIgnoreCase(caseDataPair.getKey()))
                     .findAny()
-                    .map(caseField -> validateField(caseDataPair.getKey(), caseDataPair.getValue(), caseField, fieldIdPrefix))
+                    .map(caseField -> validateField(
+                        caseDataPair.getKey(),
+                        caseDataPair.getValue(),
+                        caseField, fieldIdPrefix
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidator.java:52]: `caseField, fieldIdPrefix`
can you put this on separate lines please

**Pull request closed without merging**